samples/web/content/apprtc/apprtc.py

```diff
@@ -495,12 +515,109 @@ def write_response(self, result, params, messages):
       'result': result,
       'params': params
     }))
+    
+  def report_error(self, error):
+    self.write_response(error, {}, [])
 
   def write_room_parameters(self, room_id, client_id, messages, is_initiator):
     params = get_room_parameters(self.request, room_id, client_id, is_initiator)
     self.write_response('SUCCESS', params, messages)
+    
+  def handle_call(self, msg, room_id):
+    """Handles a new call request."""
+    if not util.has_msg_fields(msg, ((PARAM_CALLER_GCM_ID, basestring),
+        (PARAM_CALLEE_ID, basestring))):
+      self.report_error(constants.RESPONSE_INVALID_ARGUMENT)
+      return
+      
+    callerId = msg[PARAM_CALLER_GCM_ID]
+    calleeId = msg[PARAM_CALLEE_ID]
+    # Look up and validate caller by gcm id.
+    callerRecords = gcmrecord.GCMRecord.get_by_gcm_id(callerId, True)
```

Comment:
One option is to add a flag to constants.py, e.g.
ID_VERIFICATION_SUPPOORTED = False, and use this flag in such APIs calls.
So in the future we can just flip this flag when verification is added.

On Tue, Jan 27, 2015 at 2:26 PM, Chuck Hays <notifications@github.com>
wrote:

> In samples/web/content/apprtc/apprtc.py
> <https://github.com/GoogleChrome/webrtc/pull/436#discussion_r23648902>:
>
> >
> >    def write_room_parameters(self, room_id, client_id, messages, is_initiator):
> >      params = get_room_parameters(self.request, room_id, client_id, is_initiator)
> >      self.write_response('SUCCESS', params, messages)
> > +
> > +  def handle_call(self, msg, room_id):
> > +    """Handles a new call request."""
> > +    if not util.has_msg_fields(msg, ((PARAM_CALLER_GCM_ID, basestring),
> > +        (PARAM_CALLEE_ID, basestring))):
> > +      self.report_error(constants.RESPONSE_INVALID_ARGUMENT)
> > +      return
> > +
> > +    callerId = msg[PARAM_CALLER_GCM_ID]
> > +    calleeId = msg[PARAM_CALLEE_ID]
> > +    # Look up and validate caller by gcm id.
> > +    callerRecords = gcmrecord.GCMRecord.get_by_gcm_id(callerId, True)
>
> My concern is that we will forget to make this change. Just trying to
> figure out the best way to guarantee we don't overlook it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleChrome/webrtc/pull/436/files#r23648902>.
>



-- 
Jiayang Liu